src/compiler/parser.ts

```diff
@@ -2973,6 +2981,44 @@ namespace ts {
             return parseParenthesizedArrowFunctionExpressionHead(/*allowAmbiguity*/ false);
         }
 
+        function tryParseUnparenthesizedAsyncArrowFunctionExpressionWorker(): ArrowFunction {

```

Comment:
The existing function calls expressions like `x => x` `SimpleArrowFunctionExpression`. So I would use the same naming scheme to get `tryParseAsyncSimplyArrowFunctionExpression[Worker]`.